src/Server.cpp

```diff
@@ -1,41 +1,85 @@
 #include <thread>
 #include <iostream>
+#include <algorithm>
+#include <signal.h>
 
 #include "server.hpp"
+#include "exceptions.hpp"
+
+static Server *_instance;
 
 Server::Server(const int port, const size_t max_connections)
-    : _server_socket(std::make_unique<ServerSocket>(port)), _max_connections(max_connections)
+    : _server_socket(std::make_unique<ServerSocket>(port)), _max_connections(max_connections), _running(true)
 {
+    struct sigaction action;
+    action.sa_handler = server_signal_handler;
+    sigemptyset(&action.sa_mask);
+    action.sa_flags = 0;
+
+    sigaction(SIGINT, &action, nullptr);
```

Comment:
the `nullptr` here acts as a magic input params
we like to define a named `constexpr` with a meaningful name that describes its purpose
`static constexpr  struct sigaction* GIL_IS_COOL = nullptr;`